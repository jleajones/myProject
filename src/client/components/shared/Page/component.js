import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { frontloadConnect } from 'react-frontload';
import makeStyles from '@material-ui/core/styles/makeStyles';

const frontload = async ({ fetchData }) => {
  if (fetchData) {
    await fetchData();
  }
};

const useStyles = makeStyles({
  root: {
    padding: '30px 0'
  }
});

/**
 *
 * @param pageName
 * @param uuid
 * @param children
 * @param style
 * @param dispatchPageview
 * @returns {*}
 * @constructor
 */
const Page = ({ pageName, uuid, children, dispatchPageview }) => {
  const classes = useStyles();
  useEffect(() => {
    if (uuid) {
      dispatchPageview({
        pageName,
        uuid,
        url: window.location.href,
        properties: {
          referrer: document.referrer,
          screenX: window.innerWidth,
          screenY: window.innerHeight,
          userAgent: window.navigator.userAgent,
          languages: window.navigator.languages
        }
      });
    }
  }, [pageName, uuid]);
  return <main className={classes.root}>{children}</main>;
};

Page.propTypes = {
  children: PropTypes.node.isRequired,
  pageName: PropTypes.string.isRequired,
  uuid: PropTypes.string.isRequired,
  dispatchPageview: PropTypes.func.isRequired
};

export default frontloadConnect(frontload)(Page);
