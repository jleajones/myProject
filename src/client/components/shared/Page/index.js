import { connect } from 'react-redux';
import { dispatchPageview } from '@store/analytics/thunk';

import Component from './component';

const mapStateToProps = state => {
  const { analytics } = state;
  return {
    uuid: analytics.uuid
  };
};

const mapDispatchToProps = {
  dispatchPageview
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Component);
