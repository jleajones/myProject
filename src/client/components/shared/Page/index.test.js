import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import Page from '.';

describe('<Page />', () => {
  const store = configureStore();
  const pageName = 'Test Page Name';
  const dispatchPageview = jest.fn();

  const initialState = {
    analytics: {
      uuid: 'aaa-adsf-def-ejh8d'
    }
  };

  test('Renders component without crashing', () => {
    const component = shallow(
      <Page
        store={store(initialState)}
        dispatchPageview={dispatchPageview}
        pageName={pageName}
      >
        <h1>some child</h1>
      </Page>
    );

    expect(component).toMatchSnapshot();
  });
});
