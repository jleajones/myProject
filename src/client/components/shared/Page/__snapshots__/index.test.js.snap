// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> Renders component without crashing 1`] = `
<ContextProvider
  value={null}
>
  <Component
    dispatchPageview={[Function]}
    pageName="Test Page Name"
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    uuid="aaa-adsf-def-ejh8d"
  >
    <h1>
      some child
    </h1>
  </Component>
</ContextProvider>
`;
